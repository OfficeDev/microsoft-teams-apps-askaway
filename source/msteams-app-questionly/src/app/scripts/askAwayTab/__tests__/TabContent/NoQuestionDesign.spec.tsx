/**
 * @jest-environment jsdom
 */

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { Flex, Image, Text } from '@fluentui/react-northstar';
import { configure, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';
import NoQuestionDesign from '../../TabContent/NoQuestionDesign';

configure({ adapter: new enzymeAdapterReact16() });

describe('NoQuestionDesign Component', () => {
    let t;
    beforeAll(() => {
        t = jest.fn();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<NoQuestionDesign t={t} isSessionActive={true} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<NoQuestionDesign t={t} isSessionActive={true} />);

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Image)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(1);
    });
});
