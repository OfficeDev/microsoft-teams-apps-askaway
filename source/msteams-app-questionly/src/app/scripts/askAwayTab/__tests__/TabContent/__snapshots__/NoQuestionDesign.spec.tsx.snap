// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskAwayTabRemove Component should match the snapshot 1`] = `
<div>
  <div
    className="no-question-layout"
  >
    <Flex
      column={true}
    >
      <Image
        accessibility={[Function]}
        as="img"
        src="no-question.png"
      />
    </Flex>
  </div>
  <div
    className="text-center"
  >
    <Text
      as="span"
      weight="bold"
    />
  </div>
</div>
`;

exports[`NoQuestionDesign Component should match the snapshot 1`] = `
<div>
  <div
    className="no-question-layout"
  >
    <Flex
      column={true}
    >
      <Image
        accessibility={[Function]}
        as="img"
        src="no-question.png"
      />
    </Flex>
  </div>
  <div
    className="text-center"
  >
    <Text
      as="span"
      weight="bold"
    />
  </div>
</div>
`;
